packages/xo-server-usage-report/src/index.js

```diff
@@ -288,24 +322,38 @@ async function getHostsStats ({ runningHosts, xo }) {
   )
 }
 
-function getSrsStats (xoObjects) {
+async function getSrsStats ({ xo, xoObjects }) {
   return orderBy(
-    map(filter(xoObjects, obj => obj.type === 'SR' && obj.size > 0), sr => {
-      const total = sr.size / gibPower
-      const used = sr.physical_usage / gibPower
-      let name = sr.name_label
-      if (!sr.shared) {
-        name += ` (${find(xoObjects, { id: sr.$container }).name_label})`
-      }
-      return {
-        uuid: sr.uuid,
-        name,
-        total,
-        used,
-        free: total - used,
+    await asyncMap(
+      filter(
+        xoObjects,
+        obj => obj.type === 'SR' && obj.size > 0 && obj.$PBDs.length > 0
+      ),
+      async sr => {
+        const totalSpace = sr.size / gibPower
+        const usedSpace = sr.physical_usage / gibPower
+        let name = sr.name_label
+        if (!sr.shared) {
+          name += ` (${find(xoObjects, { id: sr.$container }).name_label})`
+        }
+
+        const { stats } = await xo.getXapiSrStats(sr.id, GRANULARITY)
```

Comment:
I'm starting to really fear the performance of this plugin on a big infra…

Reply:
True, maybe we should display a warning to the admin when the plugin is executed and let him the possibility to cancel or delay it.